common/src/main/java/org/apache/uniffle/common/metrics/RPCMetrics.java

```diff
@@ -18,27 +18,45 @@
 package org.apache.uniffle.common.metrics;
 
 import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 
 import com.google.common.collect.Maps;
+import com.google.common.collect.Queues;
 import io.prometheus.client.CollectorRegistry;
 import io.prometheus.client.Counter;
 import io.prometheus.client.Gauge;
 import io.prometheus.client.Summary;
 
 import org.apache.uniffle.common.util.Constants;
 import org.apache.uniffle.common.util.JavaUtils;
+import org.apache.uniffle.common.util.ThreadUtils;
 
 public abstract class RPCMetrics {
   protected boolean isRegistered = false;
   protected Map<String, Counter.Child> counterMap = JavaUtils.newConcurrentMap();
   protected Map<String, Gauge.Child> gaugeMap = JavaUtils.newConcurrentMap();
   protected Map<String, Summary.Child> transportTimeSummaryMap = JavaUtils.newConcurrentMap();
   protected Map<String, Summary.Child> processTimeSummaryMap = JavaUtils.newConcurrentMap();
+  private static final int WAIT_QUEUE_SIZE = 1000;
+  private final ExecutorService summaryObservePool;
   protected MetricsManager metricsManager;
   protected String tags;
 
   public RPCMetrics(String tags) {
     this.tags = tags;
+    BlockingQueue<Runnable> waitQueue = Queues.newLinkedBlockingQueue(WAIT_QUEUE_SIZE);
+    this.summaryObservePool =
+        new ThreadPoolExecutor(
+            2,
+            10,
+            60,
+            TimeUnit.SECONDS,
+            waitQueue,
+            ThreadUtils.getThreadFactory("SummaryObserveThreadPool"),
+            new ThreadPoolExecutor.DiscardPolicy());
```

Comment:
Why do we choose to discard? It may cause the loss of metrics.

Reply:
> Why do we choose to discard? It may cause the loss of metrics.

The metric calc is more slow than the GRPC thread produce metric.
And discard will not effect the server function, so we choose to discard it.
Or Dalao have good idea ?

Reply:
We consider use block queue, but it will block many metric.

Reply:
Block queue seems better. Discard policy will  produce the incorrect metrics.

Reply:
Ok，I changed to block queue, and it is equal to modify metric calc to async by thread pool